tests/test_dumpers.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Module tests."""
+
+from invenio_db import db
+from invenio_records.dumpers import ElasticsearchDumper
+
+from invenio_rdm_records.api import BibliographicRecord
+from invenio_rdm_records.dumpers import EDTFDumperExt
+
+
+def test_esdumper_with_edtfext(app, db, minimal_input_record):
+    """Test edft extension implementation."""
+    # Create a simple extension that adds a computed field.
+
+    dumper = ElasticsearchDumper(extensions=[EDTFDumperExt()])
```

Comment:
Minor: Use fixtures and function to reduce duplication:

```
@pytest.fixture()
def dumper():
     return ElasticsearchDumper(extensions=[EDTFDumperExt()])

def _create_edtf_record(data, datestr):
     data['metadata']['publication_date'] = date_str
     record = BibliographicRecord.create(data)
     db.session.commit()
     return record
```

Also, I think you can skip the dumper, because it's already in the ``BibliographicRecord``

Reply:
Alternatively, make a loop over different dates and expected result:

```python
[
('2021-01-01',      ('2021-01-01', '2021-01-01'))
('2021-01',         ('2021-01-01', '2021-01-31'))
('2021-01/2021-03', ('2021-01-01', '2021-03-31'))
]
```

Also, add tests for old dates:

1776
576

It's probably not going to work with years BC, but perhaps add a test to see.

Reply:
I commented out the three cases above.

Reply:
Minor: We will need to fix 1776 and 576, but somehow it's strange if EDTF doesn't support it......it's worth spending a bit more time on it, as it's definitely something which will come up (e.g. zenodo already have records from 1776)

Reply:
There is a limitation in python-edtf, it cannot parse anything lower than 1000AD. See: https://github.com/ixc/python-edtf/issues/35